### PR TITLE
Add support for postgresql protocol

### PR DESCRIPTION
**Pull request by tarvip**
With postgresql to initiate SSL-encrypted connection specific combination
of bytes must be sent to the server.

Message flow is described on following page
https://www.postgresql.org/docs/13/protocol-flow.html#id-1.10.5.7.11

And SSLRequest message format is described on
https://www.postgresql.org/docs/13/protocol-message-formats.html

The value of SSLRequest message becomes to bytes that is used in the code

### CODE REVIEW

File: prober/tcp.go [hunk 1]
```diff
@@ -107,6 +111,14 @@ var (
 				expect: "OK",
 			},
 		},
+		"postgresql": []queryResponse{
```
Comment (ribbybibby) [inline prober/tcp.go:114]:
```suggestion
		"postgres": []queryResponse{
```

One small nitpick, could we change the name of this option to `postgres` to match [the same option in `openssl`](https://www.openssl.org/docs/man1.1.1/man1/openssl-s_client.html):
> Currently, the only supported keywords are "smtp", "pop3", "imap", "ftp", "xmpp", "xmpp-server", "irc", __"postgres"__, "mysql", "lmtp", "nntp", "sieve" and "ldap".

**Review by ribbybibby — COMMENTED** [review]
This looks great, thank you so much for the contribution! I kept meaning to get around to adding this feature but never found the time.

I've left one small comment about the name of the option, other than that it looks good to me.

I'm going to spin up a postgres to check it out as soon as I can.

File: prober/tcp.go (hunk 1 above)
Comment (tarvip) [reply prober/tcp.go:114]: Yes, sure `postgres` is fine for me, the reason why I chose `postgresql` is that in connection URI it is `postgresql`: https://www.postgresql.org/docs/13/libpq-connect.html#LIBPQ-CONNSTRING

Comment (tarvip) [reply prober/tcp.go]: Updated.

**Review by ribbybibby — APPROVED** [review]
Sanity checked against a local postgres server and it looks good to me. 

Thinking about it, we should probably add some tests against an actual postgres server but that's outside the scope of this PR and applies to our other starttls targets and probers.

**Pull request merged**